src/Controller/MakefileTraceToJobTree.ts

```diff
@@ -0,0 +1,215 @@
+import { strict as assert } from 'assert'
+import { Job } from '../Job/Job'
+import { NormalJob } from '../Job/NormalJob'
+
+/**
+ * A Job and whether or not it has a dependent.
+ *
+ * Used during DAG construction.
+ */
+interface JobAndDependency {
+	job: Job
+	hasDependent: boolean
+}
+
+/**
+ * A list of target-lines and their associated command-lines.
+ *
+ * Used during trace parsing.
+ */
+interface TargetLineAndCommands {
+	targetLine: string
+	commands: string[]
+}
+
+// Regexes used when interpreting target-lines.
+const pathRegex = '[\\w\\-_. /]+'
+const targetLineRegexWithPrereqs = new RegExp(
+	`^.+: update target '(${pathRegex})' due to: (${pathRegex})$`,
+)
+const targetLineRegexWithoutPrereqs = new RegExp(
+	`^.+: target '(${pathRegex})' does not exist$`,
+)
+
+/**
+ * Reads a line from a trace containing a target.
+ * Extracts the target and any referenced prerequisites.
+ *
+ * The target line must match one of two recognized Regular Expressions.
+ *
+ * @param targetLine - a line from a trace containing a target.
+ * @returns the target and prerequisites.
+ */
+function extractInfoFromTargetLine(
```

Comment:
Possibly a weird suggestion, but could you move this function directly above the one that calls it (constructDAGFromTargetsAndCommands)? That makes it easier to read and know what is related.